OpenRA.Mods.RA2/Traits/PeriodicExplosion.cs

```diff
@@ -103,9 +103,17 @@ void ITick.Tick(Actor self)
 				var localoffset = body != null
 					? body.LocalToWorld(info.LocalOffset.Rotate(body.QuantizeOrientation(self, self.Orientation)))
 					: info.LocalOffset;
-
-				weapon.Impact(Target.FromPos(self.CenterPosition + localoffset), self,
-					self.TraitsImplementing<IFirepowerModifier>().Select(a => a.GetFirepowerModifier()).ToArray());
+
+				var args = new WarheadArgs
+				{
+					Weapon = weapon,
+					DamageModifiers = self.TraitsImplementing<IFirepowerModifier>().Select(a => a.GetFirepowerModifier()).ToArray(),
+					Source = self.CenterPosition,
+					SourceActor = self,
+					WeaponTarget = Target.FromPos(self.CenterPosition + localoffset)
+				};
+
+				weapon.Impact(Target.FromPos(self.CenterPosition + localoffset), args);

```

Comment:
We can probably re-use `args.WeaponTarget` here instead of calculating it again, can't we?